src/main/java/seedu/task/logic/LogicManager.java

```diff
@@ -20,22 +21,34 @@
 
     private final Model model;
     private final Parser parser;
+    private UndoableCommand previousCommand;
```

Comment:
Sort of breaks the flow because Logic is coupled with Model and Parser, and UndoableCommand is within logic